lnrpc/switchrpc/switch_server.go

```diff
@@ -0,0 +1,190 @@
+	if cfg.SwitchMacPath == "" {
+		cfg.SwitchMacPath = filepath.Join(
+			cfg.NetworkDir, DefaultSwitchMacFilename,
+		)
+	}
+
+	// Now that we know the full path of the switch macaroon, we can check
+	// to see if we need to create it or not. If stateless_init is set
+	// then we don't write the macaroons.
+	macFilePath := cfg.SwitchMacPath
+	if cfg.MacService != nil && !cfg.MacService.StatelessInit &&
+		!lnrpc.FileExists(macFilePath) {
+
+		log.Infof("Making macaroons for Switch RPC Server at: %v",
+			macFilePath)
+
+		// At this point, we know that the switch macaroon doesn't yet,
+		// exist, so we need to create it with the help of the main
+		// macaroon service.
+		switchMac, err := cfg.MacService.NewMacaroon(
+			context.Background(), macaroons.DefaultRootKeyID,
+			macaroonOps...,
+		)
+		if err != nil {
+			return nil, nil, err
+		}
+		switchMacBytes, err := switchMac.M().MarshalBinary()
+		if err != nil {
+			return nil, nil, err
+		}
+		err = os.WriteFile(macFilePath, switchMacBytes, 0644)
+		if err != nil {
+			_ = os.Remove(macFilePath)
+			return nil, nil, err
+		}
+	}
+
+	switchServer := &Server{
+		cfg: cfg,
+		// quit: make(chan struct{}),
+	}
+
+	return switchServer, macPermissions, nil
+}
+
+// Start launches any helper goroutines required for the Server to function.
+//
+// NOTE: This is part of the lnrpc.SubServer interface.
+func (s *Server) Start() error {
+	return nil
+}
+
+// Stop signals any active goroutines for a graceful closure.
+//
+// NOTE: This is part of the lnrpc.SubServer interface.
+func (s *Server) Stop() error {
+	return nil
+}
+
+// Name returns a unique string representation of the sub-server. This can be
+// used to identify the sub-server and also de-duplicate them.
+//
+// NOTE: This is part of the lnrpc.SubServer interface.
+func (s *Server) Name() string {
+	return subServerName
+}
+
+// RegisterWithRootServer will be called by the root gRPC server to direct a
+// sub RPC server to register itself with the main gRPC root server. Until this
+// is called, each sub-server won't be able to have
+// requests routed towards it.
+//
+// NOTE: This is part of the lnrpc.GrpcHandler interface.
+func (r *ServerShell) RegisterWithRootServer(grpcServer *grpc.Server) error {
+	// We make sure that we register it with the main gRPC server to ensure
+	// all our methods are routed properly.
+	RegisterSwitchServer(grpcServer, r)
+
+	log.Debugf("Switch RPC server successfully register with root " +
+		"gRPC server")
+
+	return nil
+}
+
+// RegisterWithRestServer will be called by the root REST mux to direct a sub
+// RPC server to register itself with the main REST mux server. Until this is
+// called, each sub-server won't be able to have requests routed towards it.
+//
+// NOTE: This is part of the lnrpc.GrpcHandler interface.
+func (r *ServerShell) RegisterWithRestServer(ctx context.Context,
+	mux *runtime.ServeMux, dest string, opts []grpc.DialOption) error {
+
```

Comment:
missing registration

Reply:
Nice catch. Originally had ignored the REST stuff and forgot to update this. Added  the registration to the commit entitled "specify REST annotations for switchrpc" - where I add `switch.yaml`.